app/src/main/java/com/protify/protifyapp/HomeActivity.kt

```diff
@@ -3,40 +3,86 @@ package com.protify.protifyapp
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
-import androidx.compose.material3.*
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.navigation.NavHostController
-import com.protify.protifyapp.features.calendar.CalendarUiModel
 import com.protify.protifyapp.features.calendar.CalendarView
-import com.protify.protifyapp.features.events.EventView
 import com.protify.protifyapp.features.login.FirebaseLoginHelper
 import com.protify.protifyapp.ui.theme.ProtifyTheme
 
 class HomeActivity {
+    var hasRan = false
         @Composable
         fun HomePage(navigateToAddEvent: () -> Unit) {
             val firestoreHelper = FirestoreHelper()
             val user = FirebaseLoginHelper().getCurrentUser()
             val context = LocalContext.current
+            var locationForecast by remember { mutableStateOf("Loading Forecast...") }
+            var distanceFromBrooklyn by remember { mutableStateOf("Loading Distance...") }
+
             Column(modifier = Modifier.fillMaxSize()) {
                 // TODO: Add a greeting based on the time of day and logged in user.
-                val greeting = "Good Morning, ${user?.email}!"
+                val greeting = "Good Morning, ${user?.email}! $locationForecast $distanceFromBrooklyn"
                 Text(
                     text = greeting,
                     modifier = Modifier.padding(16.dp),
                     style = MaterialTheme.typography.titleLarge
                 )
                 CalendarView().Calendar(navigateToAddEvent)
 
+                //Get Location Permissions
```

Comment:
Huge blocks of commented out code like this is a little worrying... Any reason why this is commented out?